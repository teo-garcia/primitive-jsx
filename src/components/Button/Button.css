.button {
  border: 0;
  border-radius: 4px;
  cursor: pointer;
  padding: 8px 16px;
}
