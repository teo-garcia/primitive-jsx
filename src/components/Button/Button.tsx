import * as React from 'react';
import styles from './Button.module.css';
import classNames from 'classnames';
import { ButtonProps } from '../../utils/types';

function Button(props: ButtonProps) {
  const {
    className,
    background = 'primary',
    color = 'background',
    fontType = 'primary',
    fontWeight = 'regular',
  } = props;

  return (
    <button
      className={classNames(
        styles.button,
        styles[`font-${fontWeight}`],
        styles[`font-${fontType}`],
        styles[`color-${color}`],
        styles[`background-${background}`],
        className
      )}
      {...props}
    />
  );
}

export { Button };
