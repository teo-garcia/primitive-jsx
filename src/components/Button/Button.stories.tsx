import * as React from 'react';
import { Meta, Story } from '@storybook/react';
import { ButtonProps } from '../../utils/types';
import { Button } from './Button';

export default {
  component: Button,
  title: 'Button',
} as Meta;

const Template: Story<ButtonProps> = (args) => <Button {...args} />;

//  Background Primary
export const BackgroundPrimary = Template.bind({});

BackgroundPrimary.args = {
  children: 'Primary Button',
  background: 'primary',
};

//  Background Secondary
export const BackgroundSecondary = Template.bind({});

BackgroundSecondary.args = {
  children: 'Secondary Button',
  background: 'secondary',
};

//  Font Type Primary
export const FontTypePrimary = Template.bind({});

FontTypePrimary.args = {
  children: 'Font Type Primary Button',
  fontType: 'primary',
};

//  Font Type Secondary
export const FontTypeSecondary = Template.bind({});

FontTypeSecondary.args = {
  children: 'Font Type Secondary Button',
  fontType: 'secondary',
};

//  Font Weight Regular
export const FontWeightRegular = Template.bind({});

FontWeightRegular.args = {
  children: 'Font Weight Regular Button',
  fontWeight: 'regular',
};

//  Font Weight Bold
export const FontWeightBold = Template.bind({});

FontWeightBold.args = {
  children: 'Font Weight Bold Button',
  fontWeight: 'bold',
};
