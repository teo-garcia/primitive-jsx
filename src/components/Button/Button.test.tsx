import * as React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import Button from './Button';

describe('<Button /> tests', function() {
  test('Should render provided children prop', function() {
    render(<Button>Hello World</Button>);
    expect(screen.getByText(/Hello World/i));
  });

  test('Should receive event handlers', async function() {
    const user = userEvent.setup();
    const mockEventHandler = jest.fn();
    render(<Button onClick={mockEventHandler}>Hello World</Button>);
    await user.click(screen.getByText(/hello world/i));
    // expect(mockEventHandler).toHaveBeenCalled();
  });
});
