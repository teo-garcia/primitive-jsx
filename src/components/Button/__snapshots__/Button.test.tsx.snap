// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> tests Should match with the component snapshot 1`] = `
<DocumentFragment>
  <button
    class="button font-regular font-primary color-background background-primary"
  >
    Hello World
  </button>
</DocumentFragment>
`;
